### PR TITLE
delete all the thunderbird things

### PR DESCRIPTION
**Pull request by eviljeff**
This deletes all the references to `amo.THUNDERBIRD` (apart from in `applications.py`, like we do with previous deprecated application support).   Mainly it's test changes where we want to verify non-Firefox application support works too, where I changed it to `amo.ANDROID`.  Possibly some of those changes are redundant, and there's quite likely more Thunderbird supporting code in there (I hope mainly in legacy frontend views that we'll delete soon anyway).

There's no db migration yet - I'm not sure what it will _need_ to delete not crash on -dev (if anything).  In any case there's a load of Addons + Versions + Files + (miscellanea) that should be cleared out soon.

### CODE REVIEW

File: src/olympia/addons/tests/test_cron.py [hunk 1]
```diff
@@ -55,11 +54,11 @@ def test_catchall(self):
     def test_appsupport(self):
         ids = Addon.objects.values_list('id', flat=True)
         cron._update_appsupport(ids)
-        assert AppSupport.objects.filter(app=amo.FIREFOX.id).count() == 4
+        assert AppSupport.objects.filter(app=amo.FIREFOX.id).count() == 3  # ??
 
         # Run it again to test deletes.
         cron._update_appsupport(ids)
-        assert AppSupport.objects.filter(app=amo.FIREFOX.id).count() == 4
+        assert AppSupport.objects.filter(app=amo.FIREFOX.id).count() == 3  # ??
```
Comment (eviljeff) [inline src/olympia/addons/tests/test_cron.py]: No idea why there's `3` now (or `4` before).

Comment (diox) [reply src/olympia/addons/tests/test_cron.py]: `base/seamonkey` fixture had an addon compatible with Firefox. So you're all good.

File: src/olympia/addons/tests/test_models.py [hunk 2]
```diff
@@ -326,7 +326,7 @@ def test_new_featured(self):
         assert f.count() == 3
         assert sorted(x.id for x in f) == (
             [2464, 7661, 15679])
-        f = Addon.objects.featured(amo.THUNDERBIRD)
+        f = Addon.objects.featured(amo.ANDROID)
```
[the remark this answers is not in the document]
Comment (eviljeff) [reply src/olympia/addons/tests/test_models.py]: ... to what?

[the remark this answers is not in the document]
Comment (eviljeff) [reply src/olympia/addons/tests/test_models.py]: oh! `f`.  I thought you meant `amo.ANDROID`

File: src/olympia/constants/categories.py [hunk 3]
```diff
@@ -170,73 +169,6 @@ def _immutable(self, *args):
             'other': StaticCategory(id=153, name=_(u'Other'), weight=333)
         }
     },
-    THUNDERBIRD.id: {
```
[the remark this answers is not in the document]
Comment (diox) [reply src/olympia/constants/categories.py]: Heh yeah now I remember adding that. It's fine then.

**Pull request merged**